docker/src/main/java/brooklyn/entity/container/docker/application/DockerServiceTypeResolver.java

```diff
@@ -72,16 +57,13 @@ public String getBrooklynType(String serviceType) {
         String imageName = Iterables.get(parts, 0);
         String imageTag = Iterables.get(parts, 1, "latest");
         log.debug("Creating Docker service entity with image {} and tag {}", imageName, imageTag);
+
+        EntitySpec<VanillaDockerApplication> spec = EntitySpec.create(VanillaDockerApplication.class);
         spec.configure(DockerAttributes.DOCKER_IMAGE_NAME, imageName);
         if (parts.size() == 2) {
             spec.configure(DockerAttributes.DOCKER_IMAGE_TAG, imageTag);
         }
-        if (resolver.getAttrs().containsKey("id")) {
```

Comment:
Leaving out the attributes makes the resolvers independent of the parser using them. It's up to the parser to decide how to configure the EntitySpec based on the plan. We can still add them if there's a good reason, but in this particular case (and all other usages) it's not needed. The logic already in place in [`DockerUtils.getContainerName`](https://github.com/brooklyncentral/clocker/blob/master/docker/src/main/java/brooklyn/entity/container/DockerUtils.java#L169) will still populate `DOCKER_CONTAINER_NAME` from `BrooklynCampConstants.PLAN_ID` (which is `id` attribute) so the existing behaviour is preserved.